anndata/_core/merge.py

```diff
@@ -245,6 +245,29 @@ def try_unifying_dtype(col: list) -> pd.core.dtypes.base.ExtensionDtype | None:
         return None
 
 
+def check_combinable_cols(cols: list[pd.Index], join: Literal["inner", "outer"]):
```

Comment:
Would be nice to get into the habit of using permissive types (would this really just work with lists?) and reduce the number of positional parameters in APIs.

Not important here, but a good habit for public APIs.

```suggestion
def check_combinable_cols(cols: Iterable[pd.Index], *, join: Literal["inner", "outer"]):
```

Reply:
I think I'd leave this for now.

Fewer imports, and it is always a list. If type checking were on, I think I'd even want to know if this was receiving a more exotic type. Like, accidentally passing just a single index, or passing a dataframe, or passing a dict.